projects/peapods-finance/index.js

```diff
@@ -1,48 +1,114 @@
+  const indexManager = chainSettings[api.chain].indexManager;
+  const peasToken = chainSettings[api.chain].peasToken;
+
+  //retrieve all available indexes
+  const allIndexes = await api.call({
+    abi: indexManagerABI,
+    target: indexManager,
+  });
+
+  //this array contains all contracts that own TVL
+  const ownersArray = [];
+
+  //this array determines which tokens to be counted as TVL, so it doesn't count the index wrapper token itself, just the underlying value and paired token in LP
+  const tokensArray = [];
+
+  //collect all index data
+  const indexStakingPools = [];
 
-const getTvl = async (api, indexes) => {
-  const ownerTokens = []
+  for (const i of allIndexes) {
+    //add index to owners
+    ownersArray.push(i.index);
+  }
 
   //get staking pool contract for index (spp)
-  const stakingPools = await api.multiCall({ abi: indexABI.lpStakingPool, calls: indexes, });
+  stakingPools = await api.multiCall({
+    abi: indexABI.lpStakingPool,
+    calls: ownersArray,
+  });
 
   //get assets this index consists of
-  const assetsResult = await api.multiCall({ abi: indexABIassets, calls: indexes, });
-  const stakingTokens = await api.multiCall({ abi: stakingPoolABI.stakingToken, calls: stakingPools, });
-
-  assetsResult.forEach((assets, i) => {
-    ownerTokens.push([assets.map(i => i.token), indexes[i]])
-  })
-  stakingTokens.forEach((stakingToken, i) => ownerTokens.push([[stakingToken], stakingPools[i]]))
-  await sumTokens2({ api, ownerTokens, blacklistedTokens: indexes, resolveLP: true, })
-  indexes.forEach(i => api.removeTokenBalance(i))
-  return api.getBalances()
+  const assetsResult = await api.multiCall({
+    abi: indexABIassets,
+    calls: ownersArray,
+  });
+  for (const ar of assetsResult) {
+    for (const asset of ar) {
+      //add asset to whitelisted tokens to be counted for tvl
+      tokensArray.push(asset.token);
+    }
+  }
+
+  //get staking token for index (UNI-V2 LP)
+  stakingTokens = await api.multiCall({
+    abi: stakingPoolABI.stakingToken,
+    calls: stakingPools,
+  });
+
+  //add UNI pool to owners
+  ownersArray.push(...stakingTokens);
+
+  //get tokens from pair
+  allToken0 = await api.multiCall({
+    abi: stakingTokenABI.token0,
+    calls: stakingTokens,
+  });
+  allToken1 = await api.multiCall({
+    abi: stakingTokenABI.token1,
+    calls: stakingTokens,
+  });
+
+  token0 = [];
+  for (const [i, value] of allToken0.entries()) {
+    token0[i] = value;
+  }
+
+  for (const [i, token1] of allToken1.entries()) {
+    //determine which token the index token is paired against and add token to whitelisted tokens to be counted for tvl
+    tokensArray.push(token1 != ownersArray[i] ? token1 : token0[i]);
+  }
+
+  //remove duplicate tokens
+  const uniqueTokensArray = [...new Set(tokensArray)];
+
+  //blacklisting ownersArray so that all index tokens won't be counted towards TVL when added to other pods to prevent double counting
+  return sumTokens2({
+    owners: ownersArray,
```

Comment:
the issue with this code is, it checks for all tokens against all owners, if there are 5 indices, it is 25 calls instead of 5. Thats why I reverted back to my code